jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/App.java

```diff
@@ -13,136 +13,24 @@
 
 package org.eclipse.jetty.deploy;
 
-import java.io.InputStream;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Properties;
-
-import org.eclipse.jetty.server.Deployable;
 import org.eclipse.jetty.server.handler.ContextHandler;
-import org.eclipse.jetty.util.FileID;
 
 /**
- * The information about an App that is managed by the {@link DeploymentManager}.
+ * An abstract App, the component that moves through the DeploymentManager.
  */
-public class App
+public interface App
```

Comment:
I think this interface should have a sub interface:

```java
interface PathApp extends App
{
    Collection<Path> getPaths();
}
```

The `ScanningAppProvider` can then produce `PathApps` and the `DefaultContextHandlerFactory` rename to `PathContextHandlerFactory` and it will only handle `PathApps`.

Reply:
That's woefully insufficient, and hyper specific to _only_ our default provider behavior.

Reply:
Not at all.  It would be specific to any App that is based on Paths.     Our "default" `AppProvider` will scan for paths war files and use some conventions to map the paths found to an App name, which could work for wars, ears, jars, xml, directories or all sorts of application bundles.  Our "default" `ContextHandlerFactory` knows how to turn a war, directory and/or XML file into a `ContextHandler` and this would be applicable to not just the way we scan for applications.